### PR TITLE
Prepare CHANGELOG for 1.0.2 release

### PR DESCRIPTION
**Pull request by amartinezfayo**
- Update CHANGELOG for 1.0.2 release

### CODE REVIEW

**Review by azdagron — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,24 @@
 # Changelog
 
+## [1.0.2] - 2021-09-02
+
+### Added
+- Experimental support for custom authorization policies based on Open Policy Agent (OPA) (#2416)
+- SPIRE Server can now be configured to emit audit logs (#2297, #2391, #2394, #2396, #2442, #2458)
+- SDS v3 API in agent now supports the SPIFFE Certificate Validator (#2435, #2460)
```
Comment (evan2645) [inline CHANGELOG.md:8]: A little bit more about what we get, and that it is related to envoy, could be good here.

```suggestion
- Envoy SDS v3 API in agent now supports the SPIFFE Certificate Validator for federated SPIFFE authentication (#2435, #2460)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,24 @@
...
+- Enhancements in the SPIRE OIDC Discovery Provider (#2404, #2453)
```
Comment (evan2645) [inline CHANGELOG.md:9]: There were a couple changes with these PRs but I think these two were the main ones.

```suggestion
- SPIRE OIDC Discovery Provider can now listen on a unix domain socket (#2404)
- SPIRE OIDC Discovery Provider now intelligently handles host headers (#2404, #2453)
```

**Review by evan2645 — COMMENTED** [review]
thanks @amartinezfayo!!

I think we need to include the configurable deprecation that was introduced in 2404. 
The other comments are just suggestions

File: CHANGELOG.md (hunk 2 above)
Comment (evan2645) [reply CHANGELOG.md:9]: @amartinezfayo pointed out that the provider already supported UDS

```suggestion
- SPIRE OIDC Discovery Provider now intelligently handles host headers (#2404, #2453)
```

**Review by azdagron — APPROVED** [review]

**Review by evan2645 — APPROVED** [review]
thanks!

**Pull request merged**